src/node-explorer-provider.ts

```diff
@@ -327,6 +214,127 @@ export class NodeExplorerProvider
     }
   }
 
+  refreshAll() {
+    this._onDidChangeTreeData.fire(undefined);
+  }
+
+  public async handleDrop(
+    target: FileExplorer | undefined,
+    sources: vscode.DataTransfer,
+    _: vscode.CancellationToken
+  ): Promise<void> {
+    if (!target) {
+      return;
+    }
+
+    let transferItem = sources.get('application/vnd.code.tree.tsFileEntry');
+    if (transferItem && transferItem.value) {
+      const source: FileExplorer = transferItem.value[0];
+      return this.moveFile(source.uri, target);
+    }
+
+    transferItem = sources.get('text/uri-list');
+    if (transferItem && transferItem.value) {
+      transferItem.value.split('\r\n').forEach((uri: string) => {
+        return this.transferFile(vscode.Uri.parse(uri), target);
+      });
+    }
+
+    sources.forEach(({ value }, mimeType) => {
+      console.log('mimeType', mimeType);
+
+      switch (mimeType) {
+        case 'application/vnd.code.tree.tsFileEntry':
+          return this.transferFile(value.uri, target);
+
+        // From the @types/vscode package: text/uri-list
+        //   A string with `toString()`ed Uris separated by `\r\n`.
+        case 'text/uri-list':
+          value.split('\r\n').forEach((uri: string) => {
+            return this.transferFile(vscode.Uri.parse(uri), target);
+          });
+          return;
+
+        default:
+          return;
+      }
+    });
+  }
+
+  /**
+   * Move a file from within a node or across nodes.
+   */
+  async moveFile(source: vscode.Uri, target: FileExplorer) {
```

Comment:
I think moveFile here indicates the unix `mv` which means the file is deleted from the source. It might be worth renaming it to `copyFile` or something similar.

Reply:
It is moving and will delete from the source after it's moved. I updated it to first verify the size from stat as a precaution.